mindsdb/integrations/handlers/ckan_handler/ckan_handler.py

```diff
@@ -1,101 +1,246 @@
+        limit = query.limit.value if query.limit else 100
+
+        resource_id = None
+        other_conditions = []
+        for condition in conditions:
+            if isinstance(condition, list) and len(condition) == 3:
+                op, col, val = condition
+                if col == "resource_id" and op == "=":
+                    resource_id = val
+                else:
+                    other_conditions.append(condition)
+
+        if not resource_id:
+            # We are handling this by design, as we want to the user to provide a resource_id
+            message = "Please provide a resource_id in your query. Example: SELECT * FROM datastore WHERE resource_id = 'your_resource_id'"
+            df = pd.DataFrame({"message": [message]})
+            return HandlerResponse(RESPONSE_TYPE.TABLE, df)
+
+        data = self.query_datastore(resource_id, other_conditions, limit)
+        return HandlerResponse(RESPONSE_TYPE.TABLE, data)
+
+    def query_datastore(self, resource_id, conditions, limit):
+        params = {"resource_id": resource_id, "limit": limit}
+
+        filters = {}
+        for condition in conditions:
+            op, col, val = condition
+            if op == "=":
+                filters[col] = val
+
+        if filters:
+            params["filters"] = filters
+
+        result = self.handler.call_ckan_api("datastore_search", params)
+
+        records = result.get("records", [])
+        return pd.DataFrame(records)
+
+    def get_columns(self):
+        return [field['id'] for field in self.fields]
+
+
+class CkanHandler(APIHandler):
     name = "ckan"
 
     def __init__(self, name=None, **kwargs):
         super().__init__(name)
-        self.type = 'ckan'
-        self.name = name
-        self.connection_args = kwargs.get('connection_data')
-        self.ckan = None
-        self.dialect = 'postgresql'
-        self.renderer = SqlalchemyRender('postgres')
         self.connection = None
         self.is_connected = False
+        self.connection_args = kwargs.get("connection_data", {})
+
+        self.datasets_table = PackageIDsTable(self)
+        self.resources_table = ResourceIDsTable(self)
+        self.datastore_table = DatastoreTable(self)
+
+        self._register_table("datasets", self.datasets_table)
+        self._register_table("resources", self.resources_table)
+        self._register_table("datastore", self.datastore_table)
+
+    def connect(self):
+        if self.is_connected:
+            return self.connection
+
+        url = self.connection_args.get("url")
+        api_key = self.connection_args.get("api_key")
+        if not url:
+            raise ValueError("CKAN URL is required")
 
-    def connect(self) -> HandlerStatusResponse:
-        """
-        Handles the connection to a CKAN remote portal instance.
-        """
-        url = self.connection_args.get('url')
         try:
-            ckan = rc(url)
+            self.connection = RemoteCKAN(url, apikey=api_key)
```

Comment:
Is this enough for checking the connection, or do we need to somehow execute some simpler action?